zio-schema-zio-test/shared/src/main/scala/zio/schema/DeriveGen.scala

```diff
@@ -0,0 +1,1034 @@
+package zio.schema
+
+import scala.collection.immutable.ListMap
+
+import zio.Chunk
+import zio.random.Random
+import zio.schema.Schema._
+import zio.test._
+
+object DeriveGen {
+
+  def gen[A](implicit schema: Schema[A]): Gen[Sized with Random, A] =
+    schema match {
+      case GenericRecord(structure)           => listMap(structure)(gen(_))
+      case Sequence(schemaA, fromChunk, _)    => Gen.chunkOf(gen(schemaA)).map(fromChunk)
+      case Schema.Enumeration(structure)      => Gen.oneOf(structure.toList.map { case (k, v) => gen(v).map(k -> _) }: _*)
+      case Transform(codec, f, _)             => gen(codec).flatMap(f(_).fold(_ => Gen.empty, Gen.const(_)))
+      case Primitive(standardType)            => PrimitiveGen(standardType).map(_.asInstanceOf[A])
+      case Optional(codec)                    => gen(codec).map(_.asInstanceOf[A])
```

Comment:
Ah, right!
I think I need to do the following instead:
```scala
case opt: Option[a] => gen(opt.codec).map(_.asInstanceOf[a])
```
thank you  Dan!

Reply:
Ah, that's right!
Thank you Dan!

Reply:
Actually, the `Gen.oneOf(gen(codec).map(Some(_)), Gen.const(None))` and `Gen.option(gen(codec))` are not exactly equal because the former is going to shrink towards `Some(_)` and the later towards `None`. I would expect shrinking to `None` though.

Reply:
@thinkharderdev What do you think?

Reply:
I think we should reuse the logic of the existing generators as much as possible so `Gen.option(gen(codec))` looks good to me.

Reply:
Agreed